docs/development/test-install.md

```diff
@@ -0,0 +1,124 @@
+# Testing the installer
```

Comment:
We should add a small top section here talking about what this is good for, with links to the CI page for the production tests, etc.